src/Prototype/Lib/Wrapped.hs

```diff
@@ -0,0 +1,53 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE KindSignatures, DataKinds #-}
+module Prototype.Lib.Wrapped
+  ( Wrapped(..)
+  , fieldNameT
+  ) where
+
+import           Data.Aeson
+import qualified Data.Text                     as T
+import           GHC.TypeLits
+import           Protolude
+import qualified Web.FormUrlEncoded            as Form
+import           Web.HttpApiData
+
+{- | A wrapped newtype is just a representation of another type, wrapped in a field at the type level. 
```

Comment:
Instead of "wrapped in a field", I think I would say something like "associated to a specific field name, e.g. when (de)serialized to(from) JSON or HTTP form data".